moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -359,13 +359,12 @@ public static JsonReader of(BufferedSource source) {
 
   /**
    * A set of strings to be chosen with {@link #selectName} or {@link #selectString}. This prepares
-   * the encoded values of the strings so they can be read directly from the input source. It cannot
-   * read arbitrary encodings of the strings: if any of a string's characters are unnecessarily
-   * escaped in the source JSON, that string will not be selected. Similarly, if the string is
-   * unquoted or uses single quotes in the source JSON, it will not be selected. Client code that
-   * uses this class should fall back to another mechanism to accommodate this possibility.
+   * the encoded values of the strings so they can be read directly from the input source.
+   * If the string is unquoted or uses single quotes in the source JSON, it will not be selected.
```

Comment:
Remove the restriction doc. Will simplify ClassJsonAdapter and EnumJsonAdapter after this is merged (to keep the pr scope small).